### PR TITLE
Chart enhancement

### PR DESCRIPTION
**Pull request by sheckathorne**
## Description
Visual and interactivity enhancement to the pie chart which can present additional information to the user and indicate the presence click events.

## Issue ticket number and link
Opened to address issue (https://github.com/USEPA/haztrak/issues/733)

## To do
The <Pie> element fires its onAnimationEnd event immediately upon rendering when React strict mode is on. The initial chart animation only works properly when strict mode is turned off. I'm looking for some assistance in solving this problem which doesn't require strict mode to be turned off to see the initial animation in development.

## Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

File: client/src/components/Charts/ManifestStatusPieChart/ManifestStatusPieChart.tsx [hunk 1]
```diff
@@ -1,43 +1,297 @@
...
+        <Legend
+          content={(props: any) =>
+            renderLegend({ ...props, handleMouseEnter, handleMouseLeave, handleClick })
+          }
+          verticalAlign="bottom"
+          height={36}
+        />
         <Pie
+          animationDuration={animationDuration}
+          onAnimationEnd={() => setChartRenderComplete(true)}
```
Comment (dpgraham4401) [inline client/src/components/Charts/ManifestStatusPieChart/ManifestStatusPieChart.tsx]: @sheckathorne thanks for your patience. Just getting around to this. 

Sounds like you already pieced this part together from the PR. The callback passed to `onAnimationEnd` is triggering a state change (`chartRenderComplete`) which causes the entire component to rerender which strict mode doesn't like. 

We can simply axe everything related to that state. If we do that, the initial animation works correctly with strict mode enabled. I'll commit and push my changes for that part. 

There's a small hiccup, the labels are added before the animation is complete. Weirdly enough, if we comment out the `<Legend/>` element, that problem goes away. I can't explain that one yet.

Comment (dpgraham4401) [reply client/src/components/Charts/ManifestStatusPieChart/ManifestStatusPieChart.tsx]: overall, I'm liking this direction this is headed. I'll stop work here and let you continue.

Comment (sheckathorne) [reply client/src/components/Charts/ManifestStatusPieChart/ManifestStatusPieChart.tsx]: Thanks for the cleanup and like you said, there is some definite unexpected behavior happening with the labels.

The problem with the labels is when you hover to activate a pie slice, then mouse away to white space, the labels don't reappear for 1500ms, equal to the default animation duration. You can combat this by setting `animationDuration` in the `Pie` element to 0, which also unfortunately removes the initial animation.

My counter to the above problem was the (admittedly unideal) `chartRenderComplete` state to set `animationDuration` to 0 after the initial animation was completed. However, problems with `onAnimationEnd` in strict mode is preventing the intended behavior.

I'm going to keep at this issue to see if there's some other solution I'm missing. Maybe the `Legend` element is the key.

Comment (dpgraham4401) [reply client/src/components/Charts/ManifestStatusPieChart/ManifestStatusPieChart.tsx]: @sheckathorne I've debugged the issue. We just needed to use both `onAnimationEnd` and `onAnimationStart` to ensure that we're keeping state up to date.

This is ready for your review.

File: client/src/components/Charts/ManifestStatusPieChart/ManifestStatusPieChart.tsx [hunk 2]
```diff
@@ -1,43 +1,299 @@
...
+      dx={dx}
+      dy={dy}
+      fill="white"
+      dominantBaseline="central"
+    >
+      {`${value}`}
+    </text>
+  );
+
+  return activeIndex !== index ? labelElement : null;
```
Comment (dpgraham4401) [inline client/src/components/Charts/ManifestStatusPieChart/ManifestStatusPieChart.tsx:83]: @sheckathorne I made one smaller other edit. The number of manifests are shown after the initial render whether the user is hovering or not. This ensures that two labels are not applied to the same pie slice.

**Review by dpgraham4401 — APPROVED** [review]

File: client/src/components/Charts/ManifestStatusPieChart/ManifestStatusPieChart.tsx (hunk 2 above)
Comment (sheckathorne) [reply client/src/components/Charts/ManifestStatusPieChart/ManifestStatusPieChart.tsx:83]: This all looks great, thank you so much for helping figure out that issue. I have no further plans for the pie chart, you can go ahead and merge.

**Pull request merged**